features/step_definitions/cluster_configuration/cluster_config.rb

```diff
@@ -33,3 +33,41 @@
   end
 end
 
+# helper method to enabled qe-app-registry
+# @args
+# 1. catalogname (default to `qe-app-registry` if none is given)
+# 2. index-version: if none is given, then just use the oc master version of the cluster
+# 3. test_type: `stage` or nil.
+Given /^I create #{QUOTED} catalogsource(?: with index version #{QUOTED})? for#{OPT_QUOTED} testing$/ do |catalog_name, index_version, test_type|
+  ensure_admin_tagged
+  ensure_destructive_tagged
+  step %Q/I store master major version in the :master_version clipboard/ unless index_version
+  index_version ||= cb.master_version
+  # find out what the test_type is,
+  test_type ||= ENV['OCP_TEST_TYPE']
+
+  project("openshift-marketplace")
+  if test_type=='stage' and index_version > "4.4"
+    iib="registry-proxy.engineering.redhat.com/rh-osbs/iib-pub-pending:v#{index_version}"
+  else
+    if %w(4.1 4.2 4.3 4.4 4.5).include? index_version
+      iib = "quay.io/openshift-qe-optional-operators/qe45-index:latest"
+    else
+      iib = "quay.io/openshift-qe-optional-operators/ocp4-index:latest"
+    end
+  end
+  logger.info("Using IIB #{iib}...")
+  cb.iib = iib
+  cb.catalog_name = catalog_name
+  step %Q/I switch to cluster admin pseudo user/
+  if catalog_source(catalog_name).exists?
+    # step %Q/I ensure "#{catalog_name}" opsrc is deleted/
+    step %Q/I ensure "#{catalog_name}" catalogsource is deleted/
+    step %Q/I ensure "#{catalog_name}" deployment is deleted/
+  end
+  # create policy
+  step %Q|I obtain test data file "catalogsource/image_content_soruce_policy.yaml"|
+  admin.cli_exec(:create, f: 'image_content_soruce_policy.yaml')
```

Comment:
Shall we use 'oc apply -f ' in case the ICSP exsit?

Reply:
@anpingli sure, I'll update it...just following your script. :)

Reply:
/lgtm